docgen/src/guides/advanced-topics.md

```diff
@@ -97,7 +97,77 @@ Here's an example
 **Location Debouncing**
  
 Since UI updates can happen in quick succession, for instance when the user types in a query, 
-the new locations are debounced. The treshold prop controls how long the component should wait between 
+the new locations are debounced. The threshold prop controls how long the component should wait between 
 state changes before pushing a new location instead of replacing the old one.
 
+## How to synchronize the url and the state with your own store and routing solution
```

Comment:
Would remove the previous topic and call this one "URL Routing" or just "Routing"